hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java

```diff
@@ -102,131 +99,101 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
-            final var fungibleTransfers = tokenTransfer.transfersOrElse(emptyList());
-            validateTrue(allowancesEnabled || !hasAllowance(fungibleTransfers), NOT_SUPPORTED);
-            totalFungibleTransfers += fungibleTransfers.size();
-
-            // Validate the nft transfer(s) (if present)
-            final var nftTransfers = tokenTransfer.nftTransfersOrElse(emptyList());
-            validateTrue(nftsEnabled || nftTransfers.isEmpty(), NOT_SUPPORTED);
-            validateTrue(allowancesEnabled || !hasNftAllowance(nftTransfers), NOT_SUPPORTED);
-            totalNftTransfers += nftTransfers.size();
-
-            // Verify that the current total number of (counted) fungible transfers does not exceed the limit
-            validateTrue(
-                    totalFungibleTransfers < ledgerConfig.tokenTransfersMaxLen(),
-                    TOKEN_TRANSFER_LIST_SIZE_LIMIT_EXCEEDED);
-            // Verify that the current total number of (counted) nft transfers does not exceed the limit
-            validateTrue(totalNftTransfers < ledgerConfig.nftTransfersMaxLen(), BATCH_SIZE_LIMIT_EXCEEDED);
+        final var steps = decomposeIntoSteps(replacedOp);
+        for (final var step : steps) {
+            // Apply all changes to the handleContext's States
+            step.doIn(transferContext);
         }
     }
 
-    private boolean hasAllowance(@NonNull final List<AccountAmount> transfers) {
-        for (final AccountAmount transfer : transfers) {
-            if (transfer.isApproval()) {
-                return true;
-            }
+    /**
+     * Ensures all aliases specified in the transfer exist. If the aliases are in receiver section, and don't exist
+     * they will be auto-created. This step populates resolved aliases and number of auto creations in the
+     * transferContext, which is used by subsequent steps and throttling.
+     * It will also replace all aliases in the {@link CryptoTransferTransactionBody} with its account ids, so it will
+     * be easier to process in next steps.
+     * @param txn the given transaction body
+     * @param transferContext the given transfer context
+     * @param context the given handle context
+     * @return the replaced transaction body with all aliases replaced with its account ids
+     * @throws HandleException if any error occurs during the process
+     */
+    private CryptoTransferTransactionBody ensureAndReplaceAliasesInOp(
+            final TransactionBody txn, final TransferContextImpl transferContext, final HandleContext context)
+            throws HandleException {
+        final var op = txn.cryptoTransferOrThrow();
+        // Ensures all aliases specified in the transfer exist
+        // If the aliases are in receiver section, and don't exist they will be auto-created
+        // This step populates resolved aliases and number of auto creations in the transferContext,
+        // which is used by subsequent steps and throttling
```

Comment:
Afaict this is the same comment in the javadoc

Reply:
Yes, removed and added other comment